tests/lib/test_irap_ascii.cpp

```diff
@@ -20,17 +22,11 @@ SCENARIO("Verify that surfio can read and write irap ascii files", "[test_irap_a
       .xrot = 0.,
       .yrot = 0.
   };
-  auto _values = std::vector<float>(_header.ncol * _header.nrow);
-  std::mt19937 g;
-  std::uniform_real_distribution<> u;
-  std::generate(_values.begin(), _values.end(), [&]() { return u(g); });
-
-  auto _original = irap{.header = _header, .values = _values};
-  export_irap_to_ascii_file("surf.irap", _original);
-  auto _imported = import_irap_ascii("surf.irap");
+  auto values = create_random_values(header.ncol * header.nrow);
+  auto original = irap{.header = header, .values = values};
+  export_irap_to_ascii_file("surf.irap", original);
+  auto imported = import_irap_ascii("surf.irap");
 
-  CHECK(_imported.header == _original.header);
-  CHECK(std::ranges::equal(_imported.values, _original.values, [](auto lhs, auto rhs) {
-    return (lhs - rhs) < 0.01;
-  }));
+  CHECK(imported.header == original.header);
+  CHECK_THAT(imported.values, Matchers::Approx(original.values).margin(0.001));
```

Comment:
Why there is a need for a margin?

Reply:
because we write it to ascii with a limited amount of precision. not sure exactly how much margin is needed, but 0.001 should be enough to pick up most errors I would think?